New package: XDiag v0.2.0

JuliaRegistrator opened the pull request:
- Registering package: XDiag
- Repository: https://github.com/awietek/XDiag.jl
- Created by: @awietek
- Version: v0.2.0
- Commit: a07af2551fb94efad57fd6aedfaedbb44429fe1c
- Reviewed by: @awietek
- Reference: https://github.com/awietek/XDiag.jl/issues/6#issuecomment-2284445517
- Description: Julia library for Exact Diagonalization of quantum many-body systems
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a83598837cf67f889976ed72f7c0ee318ec5259657da82f2462e39fdc3ea298bf95443de7483a2297a42b614200df4fb9d18a539695383a4e6bd20b0ab2e121ed172e3af44aff54e14d2b0d09eb04cc323aba2bcb3e15e215394265239ee6834836ffdb256089e66132215e587be1647b45e78 -->

awietek commented:
This is to register a new Julia package XDiag.jl. Its purpose is to perform efficient Exact Diagonalizations of quantum many body physics. This is of use in e.g. solid state physics, chemistry or quantum optics, where full many-body quantum mechanical problems are treated. The package is based on the backend C++ library xdiag, available here: https://github.com/awietek/xdiag The documentation is available under https://awietek.github.io/xdiag/ which is a combined documentation of both the C++ and Julia library.

The Julia library is OpenMP parallelized and features quantum mechanical models for spins and electrons. The C++ side of the code has already been used in several publications and an extensive testing suite is continuously tested for the C++ side. On the julia side in XDiag.jl, several tests certify the correctness of the wrapping.

The original C++ library has been wrapped using CxxWrap.jl and the binaries have been deployed via Yggdrasil, which provided the corresponding XDiag_jll package. The dependencies are kept to a minimum.

awietek commented:
From the above statement on AutoMerge, I understand that at the moment the package cannot be installed using "add XDiag". I guess this is always the case if the package is not yet registered?

JuliaRegistrator committed:
New package: XDiag v0.2.0

UUID: 3edaab83-692e-476a-bab5-541af84d25fc
Repo: https://github.com/awietek/XDiag.jl.git
Tree: a9ed525faa97642e090723c21b6ac34f811779ed

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

goerz commented:
No, that's not the problem. You're using CxxWrap as a dependency, but you didn't have a corresponding entry in your `[compat]` section. Seems like you're working on fixing it, though.

I was going to say that the name isn't very descriptive, but with it being based on https://github.com/awietek/xdiag, I suppose it's okay. Generally speaking, though, C++ package names do not make [good Julia package names](https://julialang.github.io/Pkg.jl/dev/creating-packages/#Package-naming-guidelines-1).

Anyway, as soon as you retrigger the registration with the added compat bound, you should be good to go.

[noblock]

awietek commented:
Yes, indeed there was something flawed with the [compat] section and CxxWrap. I now figured it out and it seems to work. I would like to keep the name XDiag since the C++ library is named that way and it makes sense to call both the same. 
Thank you for your comment. I am really excited to register the package and hope it'll be useful to many people.

goerz commented:
[noblock] Note that you'll have to edit all your comments to contain `[noblock]`, otherwise they will prevent the automatic merge to proceed

awietek commented:
Thanks! 

[noblock]
